### PR TITLE
fix: only show valid organizations in `CreateTemplateForm`

### LINKED ISSUE #14004: When creating a template, check authorization to determine the organizations where a user can create a template
The create template "organization dropdown" currently lists all organizations that can be read. This list would include organizations even if the actor in question is unable to create a template in said org.

The user would fill out the template form, only to receive a `401 Forbidden`. The dropdown should omit organizations that are forbidden. Forbidden in this case means the user is not allowed to create a template in the organization selected.

### PR DESCRIPTION
**Pull request by aslilac**
Closes #14004

Hide any organizations that the user might have view permissions for, but not permission to create a template

### CODE REVIEW

**Review by Emyrk — APPROVED** [review]
Permission LG :+1: 

```json
{
   "object":{
      "resource_type":"template",
      "organization_id:" <org_id>
   },
   "action":"create"
}
```

File: site/src/components/OrganizationAutocomplete/OrganizationAutocomplete.tsx [hunk 1]
```diff
@@ -51,11 +70,20 @@ export const OrganizationAutocomplete: FC<OrganizationAutocompleteProps> = ({
     750,
   );
 
+  // If an authorization check was provided, filter the organizations based on
+  // the results of that check.
+  let options = organizationsQuery.data ?? [];
+  if (check) {
```
Comment (BrunoQuaresma) [inline site/src/components/OrganizationAutocomplete/OrganizationAutocomplete.tsx:76]: What is an organization check?

**Review by BrunoQuaresma — APPROVED** [review]
Code looks good. I just have one minor question. Going to QA this right now 👀

**Comment by BrunoQuaresma** [comment]
@aslilac @Emyrk I think I found a bug with permissions. After creating a new user and adding this user into an org as admin, this user is not able to do anything related to templates.


https://github.com/user-attachments/assets/bf445870-72c5-4d5d-8900-b361db33675d

**Comment by Emyrk** [comment]
> @aslilac @Emyrk I think I found a bug with permissions. After creating a new user and adding this user into an org as admin, this user is not able to do anything related to templates.
> 
>  Screen.Recording.2024-08-06.at.12.51.31.mov

You are right, it is this call: `http://localhost:3000/api/v2/organizations/default/templates/examples`. Basically, an `org-admin` can make templates, and for some reason we query for the default template examples on the `/templates` page.

But we only query it for the primary org, which is going to fail for the org-admin in the second org.

File: site/src/components/OrganizationAutocomplete/OrganizationAutocomplete.tsx (hunk 1 above)
Comment (aslilac) [reply site/src/components/OrganizationAutocomplete/OrganizationAutocomplete.tsx:76]: if you don't pass in a check, you just get all of the organizations

**Pull request merged**